woocommerce/woocommerce
# Update ci.yml to be dispatchable and skippable

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:



This PR updates `ci.yml` such that it is skipped (except via manual dispatch) when the repo variable `SKIP_CI` is set. This allows forks that want to enable the workflow for manual testing to do so without having it run on every PR.

The PR also adds a manual dispatch, which can either be triggered for the entire suite of tests in `ci.yml`, or, if a PR number is provided, it will be triggered as if it were run on that PR.


Closes # .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a fork of this repo, and merge the changes in the PR to `trunk` within your fork.
2. Enable actions if they aren't already. You may want to disable other workflows that run on the PR so that you don't have to wait around for `ci.yml` to receive workers.
3. Create a new PR against `trunk` of your fork. Edit a file within one of the projects or plugins covered by `ci.yml`. For example, make an edit (such as adding a comment) to `packages/js/currency/src/index.ts`.
4. Verify that the project matrix is generated appropriately for the PR. If you edited the file noted in the previous step, this should run lint on currency and tests on currency and components. The tests should pass (so long as your edit didn't change anything). There will be a skipped job for handling manual dispatches.
5. Manually dispatch the `ci.yml` workflow using the PR number you created in the previous step. It should be executed in the same manner as it was from the PR, with the exception of an additional step that pulls PR information.
6. Add a repository variable to your fork from `https://github.com/YOURFORK/woocommerce/settings/variables/actions` named `SKIP_CI` with a truthy value (I use "1").
7. Create a new PR similar to step 3. This time, the CI triggered by `ci.yml` should be skipped.
8. Run the CI manually by providing the PR number from step 7. The CI should run when triggered via dispatch.


### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
